shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -131,3 +138,30 @@ - (void)dealloc {
 }
 
 @end
+
+@implementation FlutterSpellCheckResult
+
+- (instancetype)initWithMisspelledRange:(NSRange)range
+                            suggestions:(NSArray<NSString*>*)suggestions {
+  self = [super init];
+  if (self) {
+    _suggestions = suggestions;
+    _misspelledRange = range;
+  }
+  return self;
+}
+
+- (NSDictionary<NSString*, id>*)toDictionary {
+  NSMutableDictionary* result = [[[NSMutableDictionary alloc] initWithCapacity:3] autorelease];
+  result[@"location"] = @(_misspelledRange.location);
+  result[@"length"] = @(_misspelledRange.length);
+  result[@"suggestions"] = [_suggestions copy];
+  return result;
```

Comment:
❌ Failures for clang-tidy on /opt/s/w/ir/cache/builder/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm:
/opt/s/w/ir/cache/builder/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm:159:3: error: Potential leak of an object of type 'id' [clang-analyzer-osx.cocoa.RetainCount,-warnings-as-errors]
  return result;
  ^~~~~~~~~~~~~
/opt/s/w/ir/cache/builder/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm:158:28: note: Method returns an Objective-C object with a +1 retain count
  result[@"suggestions"] = [_suggestions copy];
                           ^~~~~~~~~~~~~~~~~~~
/opt/s/w/ir/cache/builder/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm:159:3: note: Object leaked: allocated object of type 'id' is not referenced later in this execution path and has a retain count of +1
  return result;
  ^~~~~~~~~~~~~